BEDS-1011/Gray-out archived dashboard rows in notifications management modal

enzo-bitfly opened the pull request.

marcel-bitfly commented on frontend/components/bc/table/BcTablePopoutEdit.vue:
```diff
@@ -4,12 +4,13 @@ import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
 
-interface Props {
+defineProps<{
+  disabled?: boolean,
```
our naming convention is `boolean props` are starting with `is-` or `has-`
Understandably the question arises what is with `DOM attributes`: 
- they could be applied via `fallthrough`
- so if we specify a `public api via our own props` we should stick to our `conventions`
```suggestion
  isDisabled?: boolean,
```

enzo-bitfly replied:
Changed to `isDisabled`

marcel-bitfly commented on frontend/components/notifications/management/NotificationsManagementDashboards.vue:
```diff
@@ -273,7 +291,9 @@ const handleDelete = (payload: Parameters<typeof deleteDashboardNotifications>[0
         >
           <template #body="slotProps">
-            {{ slotProps.data.group_name }}
+            <span :class="slotProps.data.is_archived ? 'disabled-text' : ''">
```
more concise:
```suggestion
            <span :class="{'disabled-text': slotProps.data.is_archived}>
```

enzo-bitfly replied:
Done

marcel-bitfly commented on frontend/components/bc/table/BcTablePopoutEdit.vue:
```diff
@@ -4,12 +4,13 @@ import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+defineProps<{
+  disabled?: boolean,
   label?: string,
-  noIcon?: boolean,
```
nice 💪

marcel-bitfly commented on frontend/components/notifications/management/NotificationsManagementDashboards.vue:
```diff
@@ -222,6 +222,9 @@ function getTypeIcon(type: DashboardType) {
 }
+const isDeleteButtonDisabled = (dashboard: WrappedRow) => {
+  return !dashboard.subscriptions?.length || dashboard.is_archived
+}
```
easier to reason about:
```suggestion
const isDeleteButtonDisabled = (dashboard: WrappedRow) => {
  const hasSubscriptions = dashboard.subscriptions?.length
  return !hasSubscriptions || dashboard.is_archived
}
```

enzo-bitfly replied:
Done

marcel-bitfly commented on frontend/components/notifications/management/NotificationsManagementDashboards.vue:
```diff
@@ -256,7 +259,22 @@ const handleDelete = (payload: Parameters<typeof deleteDashboardNotifications>[0
-            <span>
+            <BcTooltip
+              v-if="slotProps.data.is_archived"
+              tooltip-width="320px"
```
this might suffice: (😪 applying it by default would be best)
```suggestion
             fit-content
```

enzo-bitfly replied:
Done

marcel-bitfly commented on frontend/components/bc/table/BcTablePopoutEdit.vue:
```diff
@@ -30,12 +31,16 @@ defineProps<Props>()
-      <FontAwesomeIcon
-        v-if="!noIcon"
+      <BcButtonIcon
+        :screenreader-text="$t('common.edit')"
```
well ideally we would give more context what the button is doing, but I guess this is fine for now

marcel-bitfly commented on frontend/components/notifications/management/NotificationsManagementDashboards.vue:
```diff
@@ -256,7 +259,22 @@ const handleDelete = (payload: Parameters<typeof deleteDashboardNotifications>[0
+              tooltip-width="320px"
+              tooltip-text-align="left"
+              class="disabled-text"
+              :text="'This dashboard has been archived.'"
```
plz use a `translation` here

enzo-bitfly replied:
Done.

marcel-bitfly commented on frontend/components/notifications/management/NotificationsManagementDashboards.vue:
```diff
@@ -332,22 +356,25 @@ const handleDelete = (payload: Parameters<typeof deleteDashboardNotifications>[0
           <div class="expansion">
             <div class="info">
-              <div class="label">
+              <div :class="['label', slotProps.data.is_archived ? 'text-disabled' : '']">
```
more concise and more explicit which parts are conditional which are not
```suggestion
              <div 
              class="label"
              :class="{'text-disabled' : slotProps.data.is_archived }">
```

enzo-bitfly replied:
Done

enzo-bitfly reviewed (commented):
Thank you for the thorough review

marcel-bitfly commented on frontend/test-shh:
**issue (blocking):** plz remove this `git submodule`

enzo-bitfly replied:
Done. I finally deleted this file locally now.

marcel-bitfly commented on frontend/i18n/locales/en.json:
```diff
@@ -621,6 +621,7 @@
             "webhook": "Webhook"
         },
         "dashboards": {
+            "archived": "This dashboard has been archived.",
```
**suggestion (non-blocking):**  dunno what is better here 🤷 
```suggestion
            "is_archived": "This dashboard has been archived.",
```

marcel-bitfly replied:
in general we might look up, what good abstractions for `language files` are I guess. As I am no expert, but I think we will have a lot of trouble when adding a new language someday

enzo-bitfly replied:
I don't think we need the boolean here.
If we add more languages it may be worth looking at tools like [POEditor](https://poeditor.com/) and someone from 'product' could even edit the translations themselves through a UI.

enzo-bitfly replied:
Resolved as 'non-blocking'

enzo-bitfly committed:
feat: gray-out text and disable actions for archived dashboards in notifications management modal

See: BEDS-1101
- (BcTablePopoutEdit): improve accessibility by wrapping clickable icon with a button
- replace inexisting `custom property` used for disabled link color
